src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function adder(arg) {
+    if (arg === undefined) {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    } else {
+      sum += arg;
+
+      return adder;
+    }
+  }
+
```

Comment:
Consider adding a blank line here to separate the `adder` function definition from the return statement of `makeInfinityAdder`. This can improve readability.